### PR TITLE
Allows the unrooted_must_root lint to deal with struct variant enums, #6069

### PR DESCRIPTION
**Pull request by pgonda**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6074)

### CODE REVIEW

File: components/plugins/lints/unrooted_must_root.rs [hunk 1]
```diff
@@ -52,11 +52,19 @@ impl LintPass for UnrootedPass {
...
+            ast_map::Node::NodeItem(item) => {
+                if item.attrs.iter().all(|a| !a.check_name("must_root")) {
+                    for ref field in def.fields.iter() {
+                        lint_unrooted_ty(cx, &*field.node.ty,
+                        "Type must be rooted, use #[must_root] on the struct definition to propagate");
+                    }
+                }
             }
+            _ => {
+                let parent_id = cx.tcx.map.get_parent(id);
```
Comment (Manishearth) [inline components/plugins/lints/unrooted_must_root.rs]: We should only go up one item here.

See https://github.com/Manishearth/rust-tenacious/commit/e6d19d978698a8913f7c5f9a71670756c3add595

File: components/servo/Cargo.lock [hunk 2]
```diff
@@ -937,7 +937,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 name = "plugins"
 version = "0.0.1"
 dependencies = [
- "tenacious 0.0.1 (git+https://github.com/Manishearth/rust-tenacious.git)",
+ "tenacious 0.0.3 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (Manishearth) [inline components/servo/Cargo.lock]: Use `./mach update-cargo -p tenacious` so that all three lockfiles get updated.

**Comment by pgonda** [comment]
updated!

**Comment by Manishearth** [comment]
Review status: all files reviewed, 1 unresolved discussion, all commit checks successful.
Reviewed files:
- components/plugins/Cargo.toml @ r1
- components/plugins/lints/unrooted_must_root.rs @ r2
- components/servo/Cargo.lock @ r1
- ports/cef/Cargo.lock @ r2
- ports/gonk/Cargo.lock @ r2

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6074)

**Comment by Manishearth** [comment]
Thanks!

**Pull request merged**